server/server.go

```diff
@@ -166,21 +166,23 @@ type handler struct {
 	closed   bool
 }
 
-func (h *handler) close() {
+func (h *handler) close(err bool) {
 	if !h.closed {
 		h.conn.Close() // ignoring error: what can we do?
 		h.s.mtx.Lock()
 		delete(h.s.listeners[h.listener], h.conn)
 		h.s.mtx.Unlock()
-		logConnFailure()
 		h.closed = true
+		if err {
```

Comment:
super minor, should this func accept `err error` and this check be 
```suggestion
		if err != nil {
```
doesn't really matter though i guess

Reply:
Yea I like that, I'll update.